sdk/tables/Azure.Data.Tables/src/TableConnectionString.cs

```diff
@@ -0,0 +1,434 @@
+
+            // malformed settings string
+
+            if (settings == null)
+            {
+                accountInformation = null;
+                return false;
+            }
+
+            // helper method
+
+            string settingOrDefault(string key)
+            {
+                settings.TryGetValue(key, out var result);
+                return result;
+            }
+
+            // devstore case
+
+            if (settings.TryGetValue(TableConstants.ConnectionStrings.UseDevelopmentSetting, out var useDevSettings) && useDevSettings == "true")
+            {
+                accountInformation =
+                    settings.TryGetValue(TableConstants.ConnectionStrings.DevelopmentProxyUriSetting, out var proxyUri)
+                    ? GetDevelopmentStorageAccount(new Uri(proxyUri))
+                    : DevelopmentStorageAccount;
+
+                return true;
+            }
+
+            // non-devstore case
+            string sasToken = default;
+
+            var matchesExplicitEndpointsSpec =
+                settings.TryGetValue(TableConstants.ConnectionStrings.TableEndpointSetting, out var tableEndpoint) &&
+                Uri.IsWellFormedUriString(tableEndpoint, UriKind.Absolute) &&
+                settings.TryGetValue(TableConstants.ConnectionStrings.SharedAccessSignatureSetting, out sasToken);
+
+            var matchesAutomaticEndpointsSpec = settings.TryGetValue(TableConstants.ConnectionStrings.AccountNameSetting, out var accountName) &&
+                settings.TryGetValue(TableConstants.ConnectionStrings.AccountKeySetting, out var accountKey) &&
+                    (settings.TryGetValue(TableConstants.ConnectionStrings.TableEndpointSetting, out accountName) ||
+                    settings.TryGetValue(TableConstants.ConnectionStrings.EndpointSuffixSetting, out var endpointSuffix));
+
+            if (matchesAutomaticEndpointsSpec || matchesExplicitEndpointsSpec)
+            {
+                if (matchesAutomaticEndpointsSpec && !settings.ContainsKey(TableConstants.ConnectionStrings.DefaultEndpointsProtocolSetting))
+                {
+                    settings.Add(TableConstants.ConnectionStrings.DefaultEndpointsProtocolSetting, "https");
+                }
+
+                var tableSecondaryEndpoint = settingOrDefault(TableConstants.ConnectionStrings.TableSecondaryEndpointSetting);
+
+                // if secondary is specified, primary must also be specified
+
+                static bool s_isValidEndpointPair(string primary, string secondary) =>
+                        !string.IsNullOrWhiteSpace(primary)
+                        || /* primary is null, and... */ string.IsNullOrWhiteSpace(secondary);
+
+                (Uri, Uri) createStorageUri(string primary, string secondary, string sasToken, Func<IDictionary<string, string>, (Uri, Uri)> factory)
+                {
+                    return
```

Comment:
This expression is rather hard to read. We might be able to use pattern matching or just plain if-elses here.

Reply:
This is copied from storage - I'll take a simplification pass